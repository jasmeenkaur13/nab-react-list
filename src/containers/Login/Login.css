@media all and (min-width: 480px) {
    .Login {
      padding-bottom: 60px;
    }

    .bar{
      background-color: black;
      margin-bottom: 60px;
      height: 74px;
      padding-top: 8px;
    }
  
    .Login form {
      margin: 0 auto;
      max-width: 320px;
    }

    .Login form label {
      margin-top: 2rem;
    }

    .Login form button {
      margin-top: 2rem;
      background-color: grey;
      border-color: grey;
    }

    .Login form button:disabled {
      background-color: rgba(128, 128, 128, 0.87);
      border-color: rgba(128, 128, 128, 0.87);
    }

    .Login form button:focus,
    .Login form button:hover,
    .Login form button:focus-visible {
      background-color: grey;
      border-color: grey;
      box-shadow: none;
      outline:none;
      -webkit-tap-highlight-color: transparent;
    }

    .Login form input:focus,
    .Login form input:hover,
    .Login form input:focus-visible {
      border-color: grey;
      box-shadow: none;
      outline:none;
      -webkit-tap-highlight-color: transparent;
    }
  }