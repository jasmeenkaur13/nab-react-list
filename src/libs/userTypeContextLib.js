import { useContext, createContext } from "react";

export const UserContext = createContext('readonly');

export function useUserContext() {
  return useContext(UserContext);
}