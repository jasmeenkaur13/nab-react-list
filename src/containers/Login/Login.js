import React, { useState } from "react";
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";
import NabIcon from "../../images/logo.png"

import { useHistory } from "react-router-dom";
import "./Login.css";
import { useAppContext } from "../../libs/contextLib";
import { useUserContext } from "../../libs/userTypeContextLib";


export default function Login() {
  const { userHasAuthenticated } = useAppContext();
  const { setUserType } = useUserContext();
  const [userName, setUserName] = useState("");
  const [password, setPassword] = useState("");
  const history = useHistory();

  async function loginUser(credentials) {
    return fetch('http://localhost:3030/login', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(credentials)
    })
      .then(data => data.json())
  }

  function validateForm() {
    return userName.length > 0 && password.length > 0;
  }

  async function handleSubmit(event) {
    event.preventDefault();
    try {
      const userType = await loginUser({
        userName: userName,
        password
      });
      userHasAuthenticated(true);
      setUserType(userType.accestype);
      history.push("/dashboard");
    } catch (e) {
      alert(e.message);
    }
  }


  return (
   
    <div className="Login">
       <div className="bar">
          <img src={NabIcon} style={{height:3 +'rem', width:7 + 'rem'}}/>
        </div>
      <Form onSubmit={handleSubmit}>
        <Form.Group size="lg" controlId="userName">
          <Form.Label>UserName</Form.Label>
          <Form.Control
            autoFocus
            type="text"
            value={userName}
            onChange={(e) => setUserName(e.target.value)}
          />
        </Form.Group>
        <Form.Group size="lg" controlId="password">
          <Form.Label>Password</Form.Label>
          <Form.Control
            type="password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
          />
        </Form.Group>
        <Button block size="lg" type="submit" disabled={!validateForm()}>
          Login
        </Button>
      </Form>
    </div>
  );
}