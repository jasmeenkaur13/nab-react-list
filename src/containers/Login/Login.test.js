import React from 'react';
import renderer from 'react-test-renderer';
import Login from "./Login";
import { UserContext } from "../../libs/userTypeContextLib";
import { AppContext } from "../../libs/contextLib";

it('renders correctly initially', () => {
    const accesstype = jest.fn();
    const addItem = jest.fn();
    const tree = renderer.create(
        <AppContext.Provider value={{ addItem }}>
            <UserContext.Provider value={{ accesstype }}>
                <Login />
            </UserContext.Provider>)
    </AppContext.Provider>);
    expect(tree).toMatchSnapshot();
});



