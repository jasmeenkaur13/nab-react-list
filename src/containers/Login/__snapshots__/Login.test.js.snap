// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly initially 1`] = `
Array [
  <div
    className="Login"
  >
    <div
      className="bar"
    >
      <img
        src="logo.png"
        style={
          Object {
            "height": "3rem",
            "width": "7rem",
          }
        }
      />
    </div>
    <form
      className=""
      onSubmit={[Function]}
    >
      <div
        className="form-group"
        size="lg"
      >
        <label
          className="form-label"
          htmlFor="userName"
        >
          UserName
        </label>
        <input
          autoFocus={true}
          className="form-control"
          id="userName"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
        size="lg"
      >
        <label
          className="form-label"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="form-control"
          id="password"
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
      <button
        className="btn btn-primary btn-block btn-lg"
        disabled={true}
        type="submit"
      >
        Login
      </button>
    </form>
  </div>,
  ")",
]
`;
